Fix over aggressive stripping of relocatable files.

http://bugs.gentoo.org/183725
http://sourceware.org/bugzilla/show_bug.cgi?id=4716

2007-07-02  Alan Modra  <amodra@bigpond.net.au>
 
	PR 4716
	* objcopy.c (filter_symbols): Correct relocatable file test.

===================================================================
RCS file: /cvs/src/src/binutils/objcopy.c,v
retrieving revision 1.115
retrieving revision 1.116
diff -u -r1.115 -r1.116
--- src/binutils/objcopy.c	2007/06/29 00:34:14	1.115
+++ src/binutils/objcopy.c	2007/07/02 08:05:06	1.116
@@ -910,8 +910,7 @@
 {
   asymbol **from = isyms, **to = osyms;
   long src_count = 0, dst_count = 0;
-  int relocatable = (abfd->flags & (HAS_RELOC | EXEC_P | DYNAMIC))
-		    == HAS_RELOC;
+  int relocatable = (abfd->flags & (EXEC_P | DYNAMIC)) == 0;
 
   for (; src_count < symcount; src_count++)
     {
